src/mpi/datatype/typerep/src/typerep_util.h

```diff
@@ -159,8 +160,27 @@ static inline void BASIC_convert128(const char *src, char *dest)
 */
 #define BASIC_mixed_convert(src, dest)
 #else
-#define BASIC_convert(src, dest)               \
-        { dest = src; }
+/* FIXME: we may need use memcpy to allow no-alignment */
+static inline void BASIC_convert(const void *src, void *dest, int size)
```

Comment:
Just do `memcpy(dest, src, (size_t)size)`. Or better, why don't you just use `size_t` for the type of the `size` argument?

Reply:
For unaligned access, we may have to use memcpy or individual byte copies for the swapping. We'll do that in the slow path (for struct).

Reply:
> why don't you just use `size_t` for the type of the `size` argument?

Well, I have grown to hate `size_t` 😄  It is too easy to do arithmetics on the sizes and result in accidental underflow. A signed type is more defined than an unsigned type. On the other hand, a warning is easy to fix with no real harm.